File: app/src/main/java/com/example/kmtreader/model/enums/Transaction.java
package com.example.kmtreader.model.enums;

public enum Transaction {
    TICKET_BOOTH("LOKET"),
    TAP_OUT("TAP OUT"),
    TICKET_VENDING_MACHINE("TICKET VENDING MACHINE"),
    EXTERNAL_TRANSACTION("TRANSAKSI EKSTERNAL"),
    UNKNOWN("TIDAK DIKETAHUI");

    private String mTransactionType;

    public String getTransactionType() {
        return mTransactionType;
    }

    Transaction(String transactionType) {
        mTransactionType = transactionType;
    }

    public static Transaction getInternalTransaction(int transactionTypeCode) {
        switch (transactionTypeCode) {
            default:
                return UNKNOWN;
            case 257:
                return TAP_OUT;
            case 512:
                return TICKET_BOOTH;
            case 256:
                return TICKET_VENDING_MACHINE;
        }
    }
}

File: app/src/main/java/com/example/kmtreader/model/enums/Transaction.kt
package com.example.kmtreader.model.enums

enum class Transaction(val transactionType: String) {
    TICKET_BOOTH("LOKET"),
    TAP_OUT("TAP OUT"),
    TICKET_VENDING_MACHINE("TICKET VENDING MACHINE"),
    EXTERNAL_TRANSACTION("TRANSAKSI EKSTERNAL"),
    UNKNOWN("TIDAK DIKETAHUI");

    companion object {

        @JvmStatic
        fun getInternalTransaction(transactionTypeCode: Int): Transaction {
            return when (transactionTypeCode) {
                257 -> TAP_OUT
                512 -> TICKET_BOOTH
                256 -> TICKET_VENDING_MACHINE
                else -> UNKNOWN
            }
        }
    }
}